2/src/Item.ts

```diff
@@ -0,0 +1,40 @@
+import { Comparable } from "./Comparable";
+
+export class Item implements Comparable<Item> {
+    private static idCounter: number = 0;
```

Comment:
No implicit types. And I believe counter should start from 1.

Reply:
at this case if counter initial value will be 1 when we`ll create the first one it will be 1+1 = 2 but not 1